python/cli/convert/tum.py

```diff
@@ -0,0 +1,288 @@
+
+    def convert_distortion(model, coeffs):
+        if coeffs is None:
+            return ('pinhole', None)
+        if model == "radial-tangential":
+            c1,c2,c3,c4 = coeffs
+            return ('brown-conrady', [c1, c2, c3, c4, 0, 0, 0, 0])
+        elif model == "equidistant":
+            return ('kannala-brandt4', coeffs)
+        else:
+            raise ValueError("Unknown distortion model: " + model)
+
+    def convert_camera_model(yaml_data):
+        intrinsics = yaml_data["intrinsics"]
+        out = {
+            "focalLengthX": intrinsics[0],
+            "focalLengthY": intrinsics[1],
+            "principalPointX": intrinsics[2],
+            "principalPointY": intrinsics[3],
+            "imageWidth": yaml_data["resolution"][0],
+            "imageHeight": yaml_data["resolution"][1],
+        }
+
+        model, coeffs = convert_distortion(
+            yaml_data.get("distortion_model"),
+            yaml_data.get("distortion_coeffs", yaml_data.get("distortion_coefficients")))
+
+        if 'T_cam_imu' in yaml_data:
+            out['imuToCamera'] = yaml_data['T_cam_imu']
+        else:
+            if 'T_imu_cam' in yaml_data:
+                cam_to_imu = yaml_data['T_imu_cam']
+            elif 'T_BS' in yaml_data:
+                cam_to_imu = np.array(yaml_data['T_BS']['data']).reshape((4, 4))
+            else:
+                raise ValueError("No IMU to cam transformation found")
+            out['imuToCamera'] = np.linalg.inv(cam_to_imu).tolist()
+
+        out['model'] = model
+        out['distortionCoefficients'] = coeffs
+        return out
+
+    if os.path.exists(input_dir + "/dso/camchain.yaml"):
+        with open(input_dir + "/dso/camchain.yaml") as yamlFile:
+            data = yaml.load(yamlFile, Loader=yaml.FullLoader)
+            for i in [0, 1]:
+                cam = "cam{}".format(i)
+                d = data[cam]
+                calibration["cameras"].append(convert_camera_model(d))
+
+    elif os.path.exists(input_dir + "/mav0/cam0/sensor.yaml"):
+        for cam in ["cam0", "cam1"]:
+            with open("{}/mav0/{}/sensor.yaml".format(input_dir, cam)) as f:
+                p = yaml.load(f, Loader=yaml.FullLoader)
+                calibration["cameras"].append(convert_camera_model(p))
+    else:
+        print('Warning: no TUM calibration files found')
+        return None
+
+    return calibration
+
+def convert_with_existing_folders(rawPath, outPath, fps, crf):
+    calibration = get_calibration(rawPath)
+    if calibration is not None:
+        with open(outPath + "/calibration.json", "w") as f:
+            f.write(json.dumps(calibration, indent=2))
+
+    # The two stereo folder image files seem to be perfectly matched, unlike in the EuRoC data.
+    NS_TO_SECONDS = 1000 * 1000 * 1000 # Timestamps are in nanoseconds
+
+    # Use images that are present for both cameras.
+    # Rename bad files so that they do not match glob `*.png` given for ffmpeg.
```

Comment:
This code is quite complex. Maybe it could just give error about mismatched stereo data?